dm/worker/server.go

```diff
@@ -190,6 +199,18 @@ func (s *Server) Start() error {
 	return terror.ErrWorkerStartService.Delegate(err)
 }
 
+func (s *Server) startKeepAlive() {
+	defer s.wg.Done()
+	// worker keepalive with master
+	// If worker loses connect from master, it would stop all task and try to connect master again.
+	s.kpaCtx, s.kpaCancel = context.WithCancel(context.Background())
+	s.KeepAlive()
+}
+
+func (s *Server) stopKeepAlive() {
+	s.kpaCancel()
```

Comment:
`s.kpaCancel()` maybe called uninitialized and panic if line 151 comes too quickly. And also there is data race

Reply:
please note that you haven't fix this

Reply:
I have fix this by calling s.startKeepAlive before s.observeSourceBound